Simple Incremental/Assets/Scripts/Monobehaviours/ObjectPooler.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+/* This class is used for pooling objects and reusing objects for performance
+ * This works best when lots of the same object are spawned multiple times
+ * such as projectiles from a gun */
+public class ObjectPooler : MonoBehaviour
+{
+    public static ObjectPooler instance;
+    public Dictionary<string, Queue<GameObject>> dict = null;
+
+    void Awake()
+    {
+        if (instance == null)
+        {
+            instance = this;
+            dict = new Dictionary<string, Queue<GameObject>>();
+        }
+        else
+        {
+            Destroy(this);
+        }
+    }
+
+    public GameObject GetPooledObject(GameObject go)
+    {
+        if (!dict.ContainsKey(go.name))
+        {
+            dict.Add(go.name, new Queue<GameObject>());
+        }
+
+        if (dict[go.name].Count > 0)
+        {
+            return dict[go.name].Dequeue();
+        }
+        else
+        {
+            GameObject newGo = Instantiate(go);
+            PoolableObject po = newGo.GetComponent<PoolableObject>();
+            if( po == null)
+            {
+                po = newGo.AddComponent<PoolableObject>();
+            }
+            po.prefabName = go.name;
+            return newGo;
+        }
+
+    }
+
+    public void ReleasePooledObject(PoolableObject po)
```

Comment:
ReleasePooledObjects no longer has the ability to delete objects. If another script wants to release a PoolableObject for later re-use and there isn't already a dictionary key that matchs it, we create one and add it to this.

This allows us to have manually-placed poolableObjects in the scene that generate a pool if needed.

Reply:
The reason the ObjectPooler was deleting objects if they were not pooled objects is prefabs don't delete themselves they just disable. I moved the delete functionality out of the spawner class to the pooler. This means if someone drags and drops a prefab into the scene it will never be Destroyed it will only be disabled.